src/main/java/org/folio/list/services/refresh/ListRefreshService.java

```diff
@@ -61,22 +64,57 @@ public void doAsyncRefresh(ListEntity list, ShutdownTask shutdownTask) {
   @Async
   @Transactional(propagation = Propagation.NOT_SUPPORTED)
   public void doAsyncSorting(ListEntity list, UUID queryId, ShutdownTask shutdownTask) {
+    try (var autoCloseMe = shutdownTask) {
+      log.info("Performing async sorting for list {}, refreshId {}", list.getId(),
+        list.getInProgressRefreshId().map(UUID::toString).orElse("NONE"));
+      waitForQueryCompletion(list, queryId, shutdownTask);
+    } catch (Exception exception) {
+      log.error("Unexpected error when performing async refresh for list with id " + list.getId()
+        + ", refreshId " + (list.getInProgressRefreshId().map(UUID::toString).orElse("NONE")), exception);
+      refreshFailedCallback.accept(list, exception);
+    }
+  }
+
+  // @Async?
+  @Transactional(propagation = Propagation.NOT_SUPPORTED)
+  private void waitForQueryCompletion(ListEntity list, UUID queryId, ShutdownTask shutdownTask) throws InterruptedException {
```

Comment:
Don't think @Async is needed here since this private method is only called by other methods that have @Async annotation already